modules/ide/xsk-ide-xsaccess/src/main/resources/META-INF/dirigible/xsk-ide-xsaccess/editor/editor.js

```diff
@@ -0,0 +1,169 @@
+        $scope.data.headers.customHeaders = $scope.data.headers.customHeaders.map((x) => {
+          return {
+            ...x,
+            thevalue: x.value.substr(0, 10) == 'ALLOW-FROM' ? 'ALLOW-FROM' : x.value,
+            url: x.value.substr(0, 10) == 'ALLOW-FROM' ? x.value.substr(11) : ''
+          }
+        })
+    }
+
+    function load() {
+      getViewParameters();
+      contents = loadContents($scope.file);
+      $scope.data = JSON.parse(contents);
+      $scope.anonymous_connection_allowed = $scope.data.anonymous_connection ? true : false;
+      $scope.allAllowedMethods = ["GET", "POST", "HEAD", "OPTIONS"];
+      $scope.initJSONcontent();
+    }
+
+    load();
+
+    function updateObjectNestedProp(obj, value, propPath) {
+      const [head, ...rest] = propPath.split('.');
+      if (!rest.length) {
+        obj[head] = value;
+      } else {
+        updateObjectNestedProp(obj[head], value, rest.join('.'));
+      }
+    }
+
+    function getNestedValue(obj, path) {
+      return path.split('.').reduce(function (prev, curr) {
+        return prev[curr];
+      }, obj || this);
+    }
+
+    $scope.deleteInList = function (path, index) {
+      let newKeyValue = getNestedValue($scope.data, path).filter((_val, num) => num != index);
+      if (!newKeyValue.length) newKeyValue.push('');
+      updateObjectNestedProp(
+        $scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.addInList = function (path, value) {
+      let newKeyValue = getNestedValue($scope.data, path);
+      newKeyValue.push(value);
+      updateObjectNestedProp($scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.trigCorsMethod = function (method) {
+      if (!$scope.data.cors.allowMethods.includes(method))
+        $scope.data.cors.allowMethods.push(method);
+      else
+        $scope.data.cors.allowMethods = $scope.data.cors.allowMethods.filter((x) => x != method);
+    }
+
+    function saveContents(contentToSave) {
+      if ($scope.file) {
+        console.log('Save of XSACCESS called...');
+        let xhr = new XMLHttpRequest();
+        xhr.open('PUT', '/services/v4/ide/workspaces' + $scope.file);
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState === 4) {
+            console.log('XSACCESS file saved: ' + $scope.file);
+          }
+        };
+        xhr.send(contentToSave);
+        messageHub.post({ data: $scope.file }, 'editor.file.saved');
+        messageHub.post({ data: 'Requested file [' + $scope.file + '] saved.' }, 'status.message');
+      } else {
+        console.error('XSACCESS file parameter is not present in the request');
+      }
+    }
+
+    $scope.createSaveContent = function () {
+      let contentsJSON = JSON.parse(angular.toJson($scope.data));
+      if (contentsJSON.cors.enabled) {
+        contentsJSON.cors.allowOrigin = contentsJSON.cors.allowOrigin.filter((x) => x.trim());
```

Comment:
Why do you need to trim here? Same comment for the lines below

Reply:
We accept user input for these values, no need to save probably pasted or typed by mistake empty trailing/leading characters.